### PR TITLE
[16.07] simplify the UI of deps installation

### LINKED ISSUE #2995: improve UX for installing tool's dependencies
The admin's interface for installing dependencies has been repeatedly reported as 'unclear' and 'confusing' by admins.

### PR DESCRIPTION
**Pull request by martenson**
by hiding the details under toggle button

closes #2995

ping @jmchilton @mvdbeek @natefoo 

screenshot 2016-10-03 13 07 38
screenshot 2016-10-03 13 07 57

### CODE REVIEW

**Review by jmchilton — APPROVED** [review]
I'd prefer this target dev to be honest, but I'm okay with calling the old interface a usability bug.

**Comment by martenson** [comment]
@jmchilton It felt safe enough to target 16.07 given the nature of the changes. And this did cause serious problems to happen previously, thus `bug` label.

**Comment by nsoranzo** [comment]
This is just adding the button and hiding the details, right? Not sure it really solves the mentioned bug.

**Comment by martenson** [comment]
@nsoranzo I was hoping that together with https://github.com/galaxyproject/galaxy/pull/2998 it solves some of the problematic situations.

**Comment by nsoranzo** [comment]
@martenson I appreciate that, but looks a bit like sweeping it under the carpet...
I think we should determine which are the problem in the current interface and try to fix them, e.g.:
- the Conda part is missing a list of the packages that would be installed (I think it's the same of TS tool dependencies, but it's not clear)
- can we rephrase the text to make it clearer?
- should we include some link to the admin documentation?

**Comment by jmchilton** [comment]
@nsoranzo I think the biggest problem with the current interface is that it presents too much information - most of it irrelevant. IMO 99% of the time you install a repository, you should just install all the dependencies - you don't need to know more, you shouldn't need to care. This fix is much better than simply rephrasing and adding links.

Update: I think people who comment here are most likely following the ins and outs of conda versus tool shed, understand the difference between repository and tool dependencies, etc.... I think this form should be not for us - but for a multi-disciplinary system administrator who manages a bunch of webapps or a data hacking nerd with a Mac who wants to try out Galaxy for this or that - in both of these cases Tool Shed versus Conda is pretty irrelevant. If there is a problem, it becomes relevant and I think the other UI elements related to debugging these problems are a little better - and those should have even more information. But not here.

**Comment by martenson** [comment]
> the Conda part is missing a list of the packages that would be installed (I think it's the same of TS tool dependencies, but it's not clear)

Because we do not know at this time and for now we have no means to know (needs to be implemented). Also even if it existed this should probably not go to 16.07. As a bonus we are planning to roll out new UI for this (by @davebx ) so I would like to not touch this spooky mako codebase much.

> can we rephrase the text to make it clearer?

I am open to suggestions

> should we include some link to the admin documentation?

I added one link to dependency resolver docs

**Comment by jmchilton** [comment]
@martenson - You may want to also hide that shed tool configuration file drop down while were at it - that is even more irrelevant. It really makes it clear that we are solving a usability bug by reducing the visual clutter of irrelevant details.

**Comment by nsoranzo** [comment]
> I added one link to dependency resolver docs

We can also add one to the Conda FAQ: "When available, install externally managed dependencies (e.g. [Conda](https://docs.galaxyproject.org/en/master/admin/conda_faq.html))? <i>Beta</i>"

**Comment by martenson** [comment]
@jmchilton good call

**Comment by martenson** [comment]
@jmchilton I learned to ignore those sections so well that I do not notice it at all.

I attached it to the same button.
screenshot 2016-10-03 16 05 52

**Comment by martenson** [comment]
screenshot 2016-10-03 16 24 46

**Comment by martenson** [comment]
@galaxybot test this

**Comment by martenson** [comment]
the TS test fails are legit, working on it

**Comment by jmchilton** [comment]
Switched this to WIP until those tests are fixed. Thinking more and more we should just delay this to 16.10 - but we should definitely get this in for the release since it is a great fix to one of the more problematic parts of the admin interface.

File: templates/admin/tool_shed_repository/common.mako [hunk 1]
```diff
@@ -98,19 +115,21 @@
             revision_label_str = ' '
     %>
     <div class="form-row">
+        <p>By default Galaxy will install all needed dependencies for${revision_label_str}the repository. See the <a target="_blank" href="https://docs.galaxyproject.org/en/master/admin/dependency_resolvers.html">dependency resolver documentation</a>.</p>
+        <p>You can fine control the installation if you know what you are doing. <button class="toggle-detail-section">Display Details</button></p>
```
Comment (natefoo) [inline templates/admin/tool_shed_repository/common.mako:119]: How about "You can control how dependencies are installed (this is an advanced option, if in doubt, use the default)"

**Comment by martenson** [comment]
superseded by the PR linked above

**Pull request closed without merging**